public/app/features/dashboard/components/SubMenu/SubMenu.tsx

```diff
@@ -66,9 +68,17 @@ class SubMenuUnConnected extends PureComponent<Props> {
   }
 }
 
-const mapStateToProps: MapStateToProps<ConnectedProps, OwnProps, StoreState> = state => ({
-  variables: getSubMenuVariables(state),
-});
+const mapStateToProps: MapStateToProps<ConnectedProps, OwnProps, StoreState> = state => {
+  return {
+    variables: getSubMenuVariables(state.templating.variables),
+
+    /** Haven't found any other way to force links being passed to the links menu. Not spreading into new array makes
+     * the DashboardLinks not re-render as it gets reference to the same array in the model.
+     * Ref: https://github.com/grafana/grafana/issues/26289
+     */
+    links: [...state.dashboard.getModel()?.links],
```

Comment:
@torkelo I haven't found any other "quick" way to fix the links rendering issue and it feels hacky too. Think the mutable dashboard is problematic, as the DashboardLinks is not re-rendered, as it gets reference to the same dashboard object always. I think this is OK to force the re-render - `links` prop is used only for this and the rendering of the actual link happens based on the links from the dashboard model passed to the DashboarLinks component.

Reply:
This Will cause re render on every redux update, exactly why I did add the memorization for variables array.

Much better make parent send in links as normal prop just dashboard.links 

Then have editor create a new array in any links change

Reply:
will try this approach too

Reply:
But maybe tricky , as editor is still angular ?

Reply:
yes, unfortunately

Reply:
Another approach, with replacing links array in the model: https://github.com/grafana/grafana/pull/26311/commits/a2bb30550ce30216b22199a948171d309f693f7a

Reply:
Yea, think that is better